File: client/packages/uml-theia-integration/src/browser/diagram/diagram-configuration.ts
/********************************************************************************
 * Copyright (c) 2021 EclipseSource and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
 * available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: EPL-2.0 OR MIT
 ********************************************************************************/
import { configureDiagramServer, GLSPDiagramConfiguration } from "@eclipse-glsp/theia-integration/lib/browser";
import { createUmlDiagramContainer } from "@eclipsesource/uml-sprotty/lib";
import { Container, inject, injectable } from "inversify";

import { UmlLanguage } from "../../common/uml-language";
import {
    connectTheiaDiagramOutlineView,
    TheiaDiagramOutlineFactory,
    TheiaDiagramOutlineService
} from "../theia-diagram-outline";
import { UmlTheiaDiagramServer } from "./diagram-server";

@injectable()
export class UmlDiagramConfiguration extends GLSPDiagramConfiguration {
    @inject(TheiaDiagramOutlineFactory) protected readonly theiaDiagramOutlineFactory: () => TheiaDiagramOutlineService;

    diagramType: string = UmlLanguage.diagramType;

    doCreateContainer(widgetId: string): Container {
        const container = createUmlDiagramContainer(widgetId);

        configureDiagramServer(container, UmlTheiaDiagramServer);

        connectTheiaDiagramOutlineView(container, this.theiaDiagramOutlineFactory);

        return container;
    }
}

File: server/com.eclipsesource.uml.glsp/src/main/java/com/eclipsesource/uml/glsp/diagram/DiagramConfiguration.java
/********************************************************************************
 * Copyright (c) 2022 EclipseSource and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
 * available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: EPL-2.0 OR MIT
 ********************************************************************************/
package com.eclipsesource.uml.glsp.diagram;

import java.util.List;
import java.util.Map;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.glsp.server.types.EdgeTypeHint;
import org.eclipse.glsp.server.types.ShapeTypeHint;

public interface DiagramConfiguration {
   List<EdgeTypeHint> getEdgeTypeHints();

   List<String> getGraphContainableElements();

   List<ShapeTypeHint> getShapeTypeHints();

   Map<String, EClass> getTypeMappings();
}
